### PR TITLE
Move mouse and key actions to controller

### PR DESCRIPTION
**Pull request by ZacZhangzhuo**
### What type of change is this?

- [ ] Only move the mouse and key actions (and other relevant ones) from the `view` class to the `controller` class.
- [ ] The expandability (i.e., more actions, more key supports) is not possible yet. It will be available in the next version.

@Licini 

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas_view2/app/controller.py [hunk 1]
```diff
@@ -115,131 +235,114 @@ def glsl_version(self):
...
-
-    def view_lighted(self):
-        """Switch the view to lighted.
-
-        Returns
-        -------
-        None
-
-        """
-        self.app.view.mode = "lighted"
```
Comment (Licini) [inline src/compas_view2/app/controller.py:162]: Why are these ones deleted here?

File: src/compas_view2/app/config_default.json [hunk 2]
```diff
@@ -23,50 +23,13 @@
...
-                    },
-                    { "type": "separator" },
-                    { "type": "action", "text": "Capture", "action": "view_capture" },
-                    { "type": "separator" },
-                    { "type": "action", "text": "Front", "action": "view_front" },
-                    { "type": "action", "text": "Right", "action": "view_right" },
-                    { "type": "action", "text": "Top", "action": "view_top" },
-                    { "type": "action", "text": "Perspective", "action": "view_perspective" }
-                ]
-            }
```
Comment (Licini) [inline src/compas_view2/app/config_default.json:57]: We should still keep these as default items

File: src/compas_view2/app/config_default.json [hunk 3]
```diff
@@ -87,18 +50,6 @@
...
-                "view_front": ["control", "f2"],
-                "view_right": ["control", "f3"],
-                "view_perspective": ["control", "f4"],
-                "view_shaded": ["control", "alt", "s"],
-                "view_ghosted": ["control", "alt", "g"],
-                "view_wireframe": ["control", "alt", "w"],
-                "view_lighted": ["control", "alt", "l"],
-                "select_all": ["control", "a"],
-                "grid_show": ["f7"],
-                "view_capture": ["f2"]
```
Comment (Licini) [inline src/compas_view2/app/config_default.json:101]: same, why not keep them?

File: src/compas_view2/app/controller.py [hunk 4]
```diff
@@ -115,131 +235,114 @@ def glsl_version(self):
...
-        """Swtich to a front view.
-
-        Returns
-        -------
-        None
-
-        """
-        self.app.view.current = self.app.view.FRONT
-        self.app.view.camera.reset_position()
-        self.app.view.update_projection()
```
Comment (Licini) [inline src/compas_view2/app/controller.py:202]: same above, why removed?

**Review by Licini — CHANGES_REQUESTED** [review]
Few questions below. Also when I run an example, it raise error on the mouse due to the changes, plz make sure all examples still works accordingly : )
image

**Comment by ZacZhangzhuo** [comment]
> Few questions below. Also when I run an example, it raise error on the mouse due to the changes, plz make sure all examples still works accordingly : ) image

fixed by changing the code in `selector`:

File: src/compas_view2/app/controller.py (hunk 1 above)
Comment (ZacZhangzhuo) [reply src/compas_view2/app/controller.py]: yes! you are right actually. somehow i mixed the code at the last moment when i compareed the files. This commit should fix them!

File: src/compas_view2/app/controller.py [hunk 5]
```diff
@@ -26,8 +26,128 @@ class Controller:
...
+
+        else:
+            return False
+
+    def mouse_check(self, event, button_name: str):
+        """Giving the button name that is one of the supported buttons, return if the button is pressed.
+
+        Parameters
+        ----------
+        event:
```
Comment (Licini) [inline src/compas_view2/app/controller.py:97]: There needs to be a space after ":", otherwise the sphinx doc will not build correctly

Comment (Licini) [reply src/compas_view2/app/controller.py:97]: Also could you add class for event. same with other occurrences

Comment (ZacZhangzhuo) [reply src/compas_view2/app/controller.py]: done

**Comment by ZacZhangzhuo** [comment]
request a merge for the next step :)

**Pull request merged**